libs/ledgerjs/packages/cryptoassets/src/crypto-assets-importer/fetch/index.ts

```diff
@@ -2,10 +2,10 @@ import axios, { AxiosError } from "axios";
 
 const CRYPTO_ASSETS_BASE_URL = "https://cdn.live.ledger.com/cryptoassets";
 
-export const fetchTokens = async <T>(filename: string): Promise<T> => {
+export const fetchTokens = async <T>(filename: string): Promise<[T, string | undefined]> => {
   try {
-    const { data } = await axios.get<T>(`${CRYPTO_ASSETS_BASE_URL}/${filename}`);
-    return data;
+    const { data, headers } = await axios.get<T>(`${CRYPTO_ASSETS_BASE_URL}/${filename}`);
+    return [data, headers.etag];
```

Comment:
why not return a { data: T, etag: string } object ?

Reply:
The typing is already done by axios.
`data` is typed by the `get<T>` template, and `headers` is already an `AxiosResponseHeaders` which basically is a `Record<string, AxiosHeaders | string | string[] | number | boolean | null>`

Reply:
Yes, just asking what you think about :
 return { data, etag } instead of an array
 
 Return type would be Promise<{ data: T, etag: string | undefined }> instead of Promise<[T, string | undefined]>
 
 I think I wasn't clear enough